app/src/main/java/fr/neamar/kiss/loader/LoadContactsPojos.java

```diff
@@ -110,6 +95,33 @@ public LoadContactsPojos(Context context) {
             cur.close();
         }
 
+        // Retrieve contacts' nicknames
+        Cursor nickCursor = context.getContentResolver().query(
+                ContactsContract.Data.CONTENT_URI,
+                new String[]{
+                        ContactsContract.CommonDataKinds.Nickname.NAME,
+                        ContactsContract.Data.LOOKUP_KEY},
+                ContactsContract.Data.MIMETYPE + "= ?",
+                new String[]{ContactsContract.CommonDataKinds.Nickname.CONTENT_ITEM_TYPE},
+                null);
+
+        if (nickCursor != null) {
+            if (nickCursor.getCount() > 0) {
+                while (nickCursor.moveToNext()) {
+                    String lookupKey = nickCursor.getString(
+                            nickCursor.getColumnIndex(ContactsContract.Data.LOOKUP_KEY));
+                    String nick = nickCursor.getString(
+                            nickCursor.getColumnIndex(ContactsContract.CommonDataKinds.Nickname.NAME));
+
+                    if (nick != null && lookupKey != null) {
+                        for (ContactsPojo contact : mapContacts.get(lookupKey)) {
```

Comment:
Now, this is the only bit that could be a performance issue, since mapContacts is searched for every nickname, but I think it should be acceptable, since, as you noted before, only a small number of contacts will have a nickname.